src/json.hpp

```diff
@@ -2885,18 +2846,14 @@ class lexer
     /// the current character
     int current = std::char_traits<char>::eof();
 
-    /// whether get() should return the last character again
-    bool next_unget = false;
-
     /// the number of characters read
     std::size_t chars_read = 0;
-    /// the start position of the current token
-    std::size_t start_pos = 0;
+
+    /// raw input token string (for error messages)
+    std::vector<char> token_string { };
 
     /// buffer for variable-length tokens (numbers, strings)
-    std::vector<char> yytext = std::vector<char>(1024, '\0');
-    /// current index in yytext
-    std::size_t yylen = 0;
+    std::string yytext { };
```

Comment:
Neither of these need the `{ }` at the end.

Reply:
Correct.  However, since the -Weffc++ flag still exists (and arbitrary client code may still use it), and it is cost and risk free for us to still comply with its constraints, I decided to include the default initializer.

Reply:
I'm not sure I'd say that there is no cost.  There is a cost whenever you do something unnecessary to appease a broken tool.  It's not a runtime or build-time cost, more of a cognitive burden.  If we intend to keep it there to support the warning, anyone who maintains the code should know why it's there, which means that there should probably be a comment.
Anyway, it's all philosophical.  I'm not going to lose any sleep over it.